src/components/Drawer/Drawer.tsx

```diff
@@ -0,0 +1,234 @@
+   */
+  ariaLabelledBy?: string;
+  /**
+   * Contents of the dialog.
+   */
+  children?: React.ReactNode;
+  /**
+   * Additional class names to add to the drawer content.
+   */
+  className?: string;
+  /**
+   * Whether the drawer has a visible close button.
+   * If a title is defined, then a close button will be rendered
+   */
+  closeButton?: boolean;
+  /**
+   * If true, the drawer will close when the overlay is clicked
+   */
+  closeOnOverlayClick?: boolean;
+  /**
+   * The ref of the container where the drawer will be inserted into the DOM.
+   * By default, drawers are inserted in the document.body, but if need be they can
+   * be scoped as necessary.
+   */
+  containerRef?: React.RefObject<Node>;
+  /**
+   * By default, focus is trapped within the drawer.
+   * If true, focus will not be trapped within the contents of the drawer.
+   */
+  dangerouslyBypassFocusLock?: boolean;
+  /**
+   * By default, the drawer locks scrolling on the body, but in some cases you may want to allow for scrolling.
+   * If true, this will allow the body to scroll while the drawer is open.
+   */
+  dangerouslyBypassScrollLock?: boolean;
+  /**
+   * If true, the overlay will not be rendered, scrolling for the entire page will remain enabled,
+   * and focus will not be locked to the contents of the drawer
+   */
+  hideOverlay?: boolean;
+  /**
+   * By default the first focusable element will receive focus when the dialog
+   * opens but you can provide a ref to focus instead.
+   *
+   * @see Docs https://reach.tech/dialog#dialog-initialfocusref
+   */
+  initialFocusRef?: RefObject<HTMLDivElement>;
+  /**
+   * Which edge of the viewport should the drawer appear from
+   */
+  placement?: DrawerPlacementType;
+  /**
+   * Function that is called whenever the user either hits
+   *  the "Escape" key, clicks the close button icon, or clicks the overlay.
+   */
+  onDismiss?: (event?: React.SyntheticEvent) => void;
+  /**
+   * Title to be displayed at the top of the Drawer.
+   * A close button will be rendered automatically if this prop is defined.
+   */
+  title?: string;
+  /**
+   * The width of the Drawer when opened. Can be given a standard css value (px, rem, em, %),
+   * or a [width token](/?path=/story/design-tokens-design-tokens--page#width)
+   */
+  width?: DimensionSize | CssDimensionValue;
+}
+
+export const Drawer = forwardRef<HTMLDivElement, DrawerProps>(
+  (
+    {
+      ariaLabel = undefined,
+      ariaLabelledBy = undefined,
+      allowPinchZoom = false,
+      children = undefined,
+      className = undefined,
+      closeButton = false,
+      closeOnOverlayClick = true,
+      containerRef = undefined,
+      dangerouslyBypassFocusLock = false,
+      dangerouslyBypassScrollLock = false,
+      hideOverlay = false,
+      initialFocusRef = undefined,
+      isOpen,
+      onDismiss = undefined,
+      placement = 'right',
+      title = undefined,
+      width = undefined,
```

Comment:
we probably want to spread props here for consumer flexibility and things such as data attributes, etc. I am not sure if reach UI gives us a prop type to extend though. Maybe something we can add after an initial merge since it is not make or break.